### PR TITLE
fix: avoid pool size override

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?

Avoid pool size override

### CODE REVIEW

**Review by chasers — APPROVED** [review]

**Comment by chasers** [comment]
Approved for now, but can we have a test where the actual pool size(s) are what we are setting?

**Review by w3b6x9 — CHANGES_REQUESTED** [review]
before this goes in let's make sure `check_tenant_connection(tenant, "replication_slot_teardown")` uses just one db conn.

File: lib/realtime/database.ex [hunk 1]
```diff
@@ -98,17 +98,24 @@ defmodule Realtime.Database do
   @cdc "postgres_cdc_rls"
   @doc """
   Checks if the Tenant CDC extension information is properly configured and that we're able to query against the tenant database.
+
+  If `use_settings` is set to `false`, the connection pool will default to 1.
+  If `use_settings` is set to `true` and settings has the key `db_pool`, the connection pool will be set to the value of `db_pool`.
   """
-  @spec check_tenant_connection(Tenant.t(), binary(), integer()) ::
+  @spec check_tenant_connection(Tenant.t(), binary(), integer() | nil) ::
```
Comment (w3b6x9) [inline lib/realtime/database.ex:105]:
```suggestion
  @spec check_tenant_connection(Tenant.t(), binary(), boolean()) ::
```

**Review by w3b6x9 — APPROVED** [review]

**Pull request merged**